pkg/commands/compute/language_toolchain.go

```diff
@@ -179,6 +196,55 @@ func (bt BuildToolchain) Build() error {
 	return nil
 }
 
+// The encoding of a module starts with a preamble containing a 4-byte magic
+// number (the string '\0asm') and a version field.
+//
+// Reference:
+// https://webassembly.github.io/spec/core/binary/modules.html#binary-module
+func (bt BuildToolchain) validateWasm(path string) error {
+	// gosec flagged this:
+	// G304 (CWE-22): Potential file inclusion via variable
+	// Disabling as we trust the source of the path variable.
+	// #nosec
+	f, err := os.Open(path)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	defer f.Close()
+
+	// Parse the magic number
+	magic := make([]byte, wasmBytes)
+	_, err = f.Read(magic)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	if len(magic) > 0 && magic[0] == 0x00 {
+		magic = magic[1:] // Remove the first byte which is a null character "\x00"
+	}
+	if string(magic) != "asm" {
+		return bt.handleError(err)
+	}
```

Comment:
IMO it'd be better to declare the expected bytes here (`[]bytes{0x00, 0x61, 0x73, 0x6d}`) and using `bytes.Equal` to compare them rather than chopping the nul byte off and doing a string comparison.

Reply:
Done